geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledSelect.java

```diff
@@ -1415,11 +1413,7 @@ public boolean evaluateCq(ExecutionContext context) throws FunctionDomainExcepti
         // add UNDEFINED to results only for NOT EQUALS queries
         if (this.whereClause.getType() == COMPARISON) {
           int operator = ((Filter) this.whereClause).getOperator();
-          if ((operator != TOK_NE && operator != TOK_NE_ALT)) {
-            return false;
-          } else {
-            return true;
-          }
+          return !(operator != TOK_NE && operator != TOK_NE_ALT);
```

Comment:
That's a lot of negatives.  `return operator == TOK_NE || operator == TOK_NE_ALT` reads a lot better to me.